File: 0487-max-consecutive-ones-ii/0487-max-consecutive-ones-ii.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int ans = 0, left = 0;
        bool flip = false;
        for (int right = 0; right < nums.size(); ++right){
            if (nums[right] == 0){
                if (!flip)
                    flip = true;
                else{
                    while(left < right && nums[left] != 0)
                        ++left;
                    ++left;
                }
            }
            ans = max(ans, right - left + 1);
        }
        
        return ans;
    }
};

File: 0487-max-consecutive-ones-ii/0487-max-consecutive-ones-ii.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        ans, left, used = 0, 0, False

        for right in range(len(nums)):
            if nums[right] == 0:
                if not used:
                    used = True
                else:
                    while left < right and nums[left] != 0:
                        left += 1
                    left += 1

            ans = max(ans, right - left + 1)

        return ans